lib/shopify_app/middleware/same_site_cookie_middleware.rb

```diff
@@ -0,0 +1,58 @@
+module ShopifyApp
+  class SameSiteCookieMiddleware
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      _status, headers, _body = @app.call(env)
+    ensure
+      user_agent = env['HTTP_USER_AGENT']
+
+      if headers && headers['Set-Cookie'] && !SameSiteCookieMiddleware.same_site_none_incompatible?(user_agent) &&
+        ShopifyApp.configuration.embedded_app == true
+
+        cookies = headers['Set-Cookie'].split("\n").compact
+
+        cookies.each do |cookie|
+          headers['Set-Cookie'] = headers['Set-Cookie'].gsub("#{cookie}", "#{cookie}; secure; SameSite=None")
```

Comment:
It might be good to check here if the existing `cookie` value contains the phrase `; SameSite` as a quick hack to ensure we don't double up or trounce on intentionally secured (i.e. `SameSite=(lax|strict)`) cookies.

Reply:
Whoops, forgot this comment - just added a check for that